deepinv/loss/r2r.py

```diff
@@ -1,155 +1,230 @@
-        over multiple realizations of the added noise, i.e. :math:`\hat{x} = \frac{1}{N}\sum_{i=1}^N R(y+\alpha z_i)`
+        over multiple realizations of the added noise, i.e. :math:`\hat{x} = \frac{1}{N}\sum_{i=1}^N R(y_1^{(i)})`,
         where :math:`N>1`. This can be achieved using :meth:`adapt_model`.
 
-    :param Metric, torch.nn.Module metric: metric for calculating loss, defaults to MSE.
-    :param float sigma: standard deviation of the Gaussian noise used for the perturbation.
-    :param float alpha: scaling factor of the perturbation.
-    :param int eval_n_samples: number of samples used for the Monte Carlo approximation.
+    .. deprecated:: 0.2.3
+
+        The ``sigma`` paramater is deprecated and will be removed in future versions. Use ``noise_model`` parameter instead.
+
+    :param Metric, torch.nn.Module metric: Metric for calculating loss, defaults to MSE.
+    :param NoiseModel noise_model: Noise model of the natural exponential family, defaults to Gaussian.
+    :param float alpha: Scaling factor of the corruption.
+    :param int eval_n_samples: Number of samples used for the Monte Carlo approximation.
 
     |sep|
 
     :Example:
 
-        >>> import torch
-        >>> import deepinv as dinv
-        >>> sigma = 0.1
-        >>> physics = dinv.physics.Denoising(dinv.physics.GaussianNoise(sigma))
-        >>> model = dinv.models.MedianFilter()
-        >>> loss = dinv.loss.R2RLoss(sigma=sigma, eval_n_samples=2)
-        >>> model = loss.adapt_model(model) # important step!
-        >>> x = torch.ones((1, 1, 8, 8))
-        >>> y = physics(x)
-        >>> x_net = model(y, physics, update_parameters=True) # save extra noise in forward pass
-        >>> l = loss(x_net, y, physics, model)
-        >>> print(l.item() > 0)
-        True
+    >>> import torch
+    >>> import deepinv as dinv
+    >>> sigma = 0.1
+    >>> noise_model = dinv.physics.GaussianNoise(sigma)
+    >>> physics = dinv.physics.Denoising(noise_model)
+    >>> model = dinv.models.MedianFilter()
+    >>> loss = dinv.loss.R2RLoss(noise_model=noise_model, eval_n_samples=2)
+    >>> model = loss.adapt_model(model) # important step!
+    >>> x = torch.ones((1, 1, 8, 8))
+    >>> y = physics(x)
+    >>> x_net = model(y, physics, update_parameters=True) # save extra noise in forward pass
+    >>> l = loss(x_net, y, physics, model)
+    >>> print(l.item() > 0)
+    True
 
     """
 
     def __init__(
         self,
         metric: Union[Metric, torch.nn.Module] = torch.nn.MSELoss(),
-        sigma=0.1,
+        noise_model: NoiseModel = GaussianNoise(0.1),
         alpha=0.5,
+        sigma=None,
         eval_n_samples=5,
     ):
         super(R2RLoss, self).__init__()
-        self.name = "r2r"
+        self.name = "gr2r"
         self.metric = metric
-        self.sigma = sigma
         self.alpha = alpha
         self.eval_n_samples = eval_n_samples
+        self.noise_model = noise_model
+
+        if sigma is not None:
+
+            print(
```

Comment:
add at the beginning from `warnings import warn`
```suggestion
            warnings.warn( ...)
```
instead of the print

Reply:
thanks, update in [79f1cc3]